### ISSUE TITLE
UnionLib Crashed `1.20.1`

### ISSUE THREAD

**Issue by kennarddh** [Stereowalker/Survive#298]
I'm using `UnionLib-1.20.1-10.1.1-Forge.jar` and `Survive-1.20.1-9.0.3.jar`

Minecraft version `1.20.1`

```
[13Sep2023 00:21:02.776] [modloading-worker-0/ERROR] [net.minecraftforge.fml.loading.RuntimeDistCleaner/DISTXFORM]: Attempted to load class net/minecraft/client/Options for invalid dist DEDICATED_SERVER
--
987 | [13Sep2023 00:21:02.782] [modloading-worker-0/ERROR] [net.minecraftforge.fml.javafmlmod.FMLModContainer/LOADING]: Failed to create mod instance. ModID: unionlib, class com.stereowalker.unionlib.UnionLib
988 | java.lang.reflect.InvocationTargetException: null
989 | at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
990 | at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[?:?]
991 | at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[?:?]
992 | at java.lang.reflect.Constructor.newInstanceWithCaller(Unknown Source) ~[?:?]
993 | at java.lang.reflect.Constructor.newInstance(Unknown Source) ~[?:?]
994 | at net.minecraftforge.fml.javafmlmod.FMLModContainer.constructMod(FMLModContainer.java:68) ~[javafmllanguage-1.20.1-47.1.46.jar%23145!/:?]
995 | at net.minecraftforge.fml.ModContainer.lambda$buildTransitionHandler$10(ModContainer.java:123) ~[fmlcore-1.20.1-47.1.46.jar%23144!/:?]
996 | at java.util.concurrent.CompletableFuture$AsyncRun.run(Unknown Source) ~[?:?]
997 | at java.util.concurrent.CompletableFuture$AsyncRun.exec(Unknown Source) ~[?:?]
998 | at java.util.concurrent.ForkJoinTask.doExec(Unknown Source) ~[?:?]
999 | at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source) ~[?:?]
1000 | at java.util.concurrent.ForkJoinPool.scan(Unknown Source) ~[?:?]
1001 | at java.util.concurrent.ForkJoinPool.runWorker(Unknown Source) ~[?:?]
1002 | at java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source) ~[?:?]
1003 | Caused by: java.lang.RuntimeException: Attempted to load class net/minecraft/client/Options for invalid dist DEDICATED_SERVER
1004 | at net.minecraftforge.fml.loading.RuntimeDistCleaner.processClassWithFlags(RuntimeDistCleaner.java:57) ~[fmlloader-1.20.1-47.1.46.jar%2369!/:1.0]
1005 | at cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins(LaunchPluginHandler.java:88) ~[modlauncher-10.0.9.jar%2355!/:?]
1006 | at cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:120) ~[modlauncher-10.0.9.jar%2355!/:?]
1007 | at cpw.mods.modlauncher.TransformingClassLoader.maybeTransformClassBytes(TransformingClassLoader.java:50) ~[modlauncher-10.0.9.jar%2355!/:?]
1008 | at cpw.mods.cl.ModuleClassLoader.readerToClass(ModuleClassLoader.java:113) ~[securejarhandler-2.1.10.jar:?]
1009 | at cpw.mods.cl.ModuleClassLoader.lambda$findClass$15(ModuleClassLoader.java:219) ~[securejarhandler-2.1.10.jar:?]
1010 | at cpw.mods.cl.ModuleClassLoader.loadFromModule(ModuleClassLoader.java:229) ~[securejarhandler-2.1.10.jar:?]
1011 | at cpw.mods.cl.ModuleClassLoader.findClass(ModuleClassLoader.java:219) ~[securejarhandler-2.1.10.jar:?]
1012 | at cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:135) ~[securejarhandler-2.1.10.jar:?]
1013 | at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:?]
1014 | at java.lang.Class.getDeclaredConstructors0(Native Method) ~[?:?]
1015 | at java.lang.Class.privateGetDeclaredConstructors(Unknown Source) ~[?:?]
1016 | at java.lang.Class.getConstructor0(Unknown Source) ~[?:?]
1017 | at java.lang.Class.getConstructor(Unknown Source) ~[?:?]
1018 | at net.minecraftforge.eventbus.api.EventListenerHelper.computeListenerList(EventListenerHelper.java:75) ~[eventbus-6.0.5.jar%2352!/:?]
1019 | at net.minecraftforge.eventbus.api.EventListenerHelper.lambda$getListenerListInternal$0(EventListenerHelper.java:56) ~[eventbus-6.0.5.jar%2352!/:?]
1020 | at net.minecraftforge.eventbus.LockHelper.computeIfAbsent(LockHelper.java:61) ~[eventbus-6.0.5.jar%2352!/:?]
1021 | at net.minecraftforge.eventbus.LockHelper.get(LockHelper.java:44) ~[eventbus-6.0.5.jar%2352!/:?]
1022 | at net.minecraftforge.eventbus.LockHelper.computeIfAbsent(LockHelper.java:39) ~[eventbus-6.0.5.jar%2352!/:?]
1023 | at net.minecraftforge.eventbus.api.EventListenerHelper.getListenerListInternal(EventListenerHelper.java:56) ~[eventbus-6.0.5.jar%2352!/:?]
1024 | at net.minecraftforge.eventbus.api.EventListenerHelper.getListenerList(EventListenerHelper.java:50) ~[eventbus-6.0.5.jar%2352!/:?]
1025 | at net.minecraftforge.eventbus.EventBus.addToListeners(EventBus.java:276) ~[eventbus-6.0.5.jar%2352!/:?]
1026 | at net.minecraftforge.eventbus.EventBus.addListener(EventBus.java:252) ~[eventbus-6.0.5.jar%2352!/:?]
1027 | at net.minecraftforge.eventbus.EventBus.addListener(EventBus.java:244) ~[eventbus-6.0.5.jar%2352!/:?]
1028 | at net.minecraftforge.eventbus.EventBus.addListener(EventBus.java:199) ~[eventbus-6.0.5.jar%2352!/:?]
1029 | at net.minecraftforge.eventbus.EventBus.addListener(EventBus.java:193) ~[eventbus-6.0.5.jar%2352!/:?]
1030 | at net.minecraftforge.eventbus.EventBus.addListener(EventBus.java:187) ~[eventbus-6.0.5.jar%2352!/:?]
1031 | at com.stereowalker.unionlib.forge.ForgeMod.clientStuff(ForgeMod.java:50) ~[UnionLib-1.20.1-10.1.1-Forge.jar%23140!/:10.1.1]
1032 | at com.stereowalker.unionlib.mod.ModHandler.registerMod(ModHandler.java:170) ~[UnionLib-1.20.1-10.1.1-Forge.jar%23140!/:10.1.1]
1033 | at com.stereowalker.unionlib.mod.MinecraftMod.<init>(MinecraftMod.java:62) ~[UnionLib-1.20.1-10.1.1-Forge.jar%23140!/:10.1.1]
1034 | at com.stereowalker.unionlib.mod.MinecraftMod.<init>(MinecraftMod.java:94) ~[UnionLib-1.20.1-10.1.1-Forge.jar%23140!/:10.1.1]
1035 | at com.stereowalker.unionlib.UnionLib.<init>(UnionLib.java:84) ~[UnionLib-1.20.1-10.1.1-Forge.jar%23140!/:10.1.1]
1036 | ... 14 more
```

```
[13Sep2023 00:21:05.126] [modloading-worker-0/ERROR] [net.minecraftforge.fml.loading.RuntimeDistCleaner/DISTXFORM]: Attempted to load class net/minecraft/client/Options for invalid dist DEDICATED_SERVER
--
1087 | [13Sep2023 00:21:05.126] [modloading-worker-0/ERROR] [net.minecraftforge.fml.javafmlmod.FMLModContainer/LOADING]: Failed to create mod instance. ModID: survive, class com.stereowalker.survive.Survive
1088 | java.lang.reflect.InvocationTargetException: null
1089 | at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
1090 | at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[?:?]
1091 | at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[?:?]
1092 | at java.lang.reflect.Constructor.newInstanceWithCaller(Unknown Source) ~[?:?]
1093 | at java.lang.reflect.Constructor.newInstance(Unknown Source) ~[?:?]
1094 | at net.minecraftforge.fml.javafmlmod.FMLModContainer.constructMod(FMLModContainer.java:68) ~[javafmllanguage-1.20.1-47.1.46.jar%23145!/:?]
1095 | at net.minecraftforge.fml.ModContainer.lambda$buildTransitionHandler$10(ModContainer.java:123) ~[fmlcore-1.20.1-47.1.46.jar%23144!/:?]
1096 | at java.util.concurrent.CompletableFuture$AsyncRun.run(Unknown Source) ~[?:?]
1097 | at java.util.concurrent.CompletableFuture$AsyncRun.exec(Unknown Source) ~[?:?]
1098 | at java.util.concurrent.ForkJoinTask.doExec(Unknown Source) ~[?:?]
1099 | at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source) ~[?:?]
1100 | at java.util.concurrent.ForkJoinPool.scan(Unknown Source) ~[?:?]
1101 | at java.util.concurrent.ForkJoinPool.runWorker(Unknown Source) ~[?:?]
1102 | at java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source) ~[?:?]
1103 | Caused by: java.lang.RuntimeException: Attempted to load class net/minecraft/client/Options for invalid dist DEDICATED_SERVER
1104 | at net.minecraftforge.fml.loading.RuntimeDistCleaner.processClassWithFlags(RuntimeDistCleaner.java:57) ~[fmlloader-1.20.1-47.1.46.jar%2369!/:1.0]
1105 | at cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins(LaunchPluginHandler.java:88) ~[modlauncher-10.0.9.jar%2355!/:?]
1106 | at cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:120) ~[modlauncher-10.0.9.jar%2355!/:?]
1107 | at cpw.mods.modlauncher.TransformingClassLoader.maybeTransformClassBytes(TransformingClassLoader.java:50) ~[modlauncher-10.0.9.jar%2355!/:?]
1108 | at cpw.mods.cl.ModuleClassLoader.readerToClass(ModuleClassLoader.java:113) ~[securejarhandler-2.1.10.jar:?]
1109 | at cpw.mods.cl.ModuleClassLoader.lambda$findClass$15(ModuleClassLoader.java:219) ~[securejarhandler-2.1.10.jar:?]
1110 | at cpw.mods.cl.ModuleClassLoader.loadFromModule(ModuleClassLoader.java:229) ~[securejarhandler-2.1.10.jar:?]
1111 | at cpw.mods.cl.ModuleClassLoader.findClass(ModuleClassLoader.java:219) ~[securejarhandler-2.1.10.jar:?]
1112 | at cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:135) ~[securejarhandler-2.1.10.jar:?]
1113 | at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:?]
1114 | at java.lang.Class.getDeclaredConstructors0(Native Method) ~[?:?]
1115 | at java.lang.Class.privateGetDeclaredConstructors(Unknown Source) ~[?:?]
1116 | at java.lang.Class.getConstructor0(Unknown Source) ~[?:?]
1117 | at java.lang.Class.getConstructor(Unknown Source) ~[?:?]
1118 | at net.minecraftforge.eventbus.api.EventListenerHelper.computeListenerList(EventListenerHelper.java:75) ~[eventbus-6.0.5.jar%2352!/:?]
1119 | at net.minecraftforge.eventbus.api.EventListenerHelper.lambda$getListenerListInternal$0(EventListenerHelper.java:56) ~[eventbus-6.0.5.jar%2352!/:?]
1120 | at net.minecraftforge.eventbus.LockHelper.computeIfAbsent(LockHelper.java:61) ~[eventbus-6.0.5.jar%2352!/:?]
1121 | at net.minecraftforge.eventbus.LockHelper.get(LockHelper.java:44) ~[eventbus-6.0.5.jar%2352!/:?]
1122 | at net.minecraftforge.eventbus.LockHelper.computeIfAbsent(LockHelper.java:39) ~[eventbus-6.0.5.jar%2352!/:?]
1123 | at net.minecraftforge.eventbus.api.EventListenerHelper.getListenerListInternal(EventListenerHelper.java:56) ~[eventbus-6.0.5.jar%2352!/:?]
1124 | at net.minecraftforge.eventbus.api.EventListenerHelper.getListenerList(EventListenerHelper.java:50) ~[eventbus-6.0.5.jar%2352!/:?]
1125 | at net.minecraftforge.eventbus.EventBus.addToListeners(EventBus.java:276) ~[eventbus-6.0.5.jar%2352!/:?]
1126 | at net.minecraftforge.eventbus.EventBus.addListener(EventBus.java:252) ~[eventbus-6.0.5.jar%2352!/:?]
1127 | at net.minecraftforge.eventbus.EventBus.addListener(EventBus.java:244) ~[eventbus-6.0.5.jar%2352!/:?]
1128 | at net.minecraftforge.eventbus.EventBus.addListener(EventBus.java:199) ~[eventbus-6.0.5.jar%2352!/:?]
1129 | at net.minecraftforge.eventbus.EventBus.addListener(EventBus.java:193) ~[eventbus-6.0.5.jar%2352!/:?]
1130 | at net.minecraftforge.eventbus.EventBus.addListener(EventBus.java:187) ~[eventbus-6.0.5.jar%2352!/:?]
1131 | at com.stereowalker.unionlib.forge.ForgeMod.clientStuff(ForgeMod.java:50) ~[UnionLib-1.20.1-10.1.1-Forge.jar%23140!/:10.1.1]
1132 | at com.stereowalker.unionlib.mod.ModHandler.registerMod(ModHandler.java:170) ~[UnionLib-1.20.1-10.1.1-Forge.jar%23140!/:10.1.1]
1133 | at com.stereowalker.unionlib.mod.MinecraftMod.<init>(MinecraftMod.java:62) ~[UnionLib-1.20.1-10.1.1-Forge.jar%23140!/:10.1.1]
1134 | at com.stereowalker.unionlib.mod.MinecraftMod.<init>(MinecraftMod.java:94) ~[UnionLib-1.20.1-10.1.1-Forge.jar%23140!/:10.1.1]
1135 | at com.stereowalker.survive.Survive.<init>(Survive.java:131) ~[Survive-1.20.1-9.0.3.jar%23136!/:1.20.1-9.0.3]
1136 | ... 14 more
```

**Comment by Stereowalker** [comment]
Resolved. Thank you for the detailed report

**Comment by Yuuzinhozi** [comment]
> Resolved. Thank you for the detailed report

resolved? im in version 1.20.1 and still with the problem :(
